adlfs/core.py

```diff
@@ -743,6 +743,9 @@ def _put_data(rest, op, path, data, max_attempts=10, **kwargs):
         except (PermissionError, FileNotFoundError):
             raise
         except Exception as e:
+            if '"BadOffsetException"' in repr(e):
```

Comment:
The two modes for opening a file via the ADLFileSystem will never allow this situation to occur (you either clobber the file to zero, or append at the end; there is no seek in write mode).
Putting the except in the uploader is more tricky, because first this function would need to retry the requisite number of times before control is passed back.